### PR TITLE
Return 0 balances everywhere

### PR DESCRIPTION
**Pull request by severinwullschleger**
getBalance should always return an array at least containing the mainAsset of a chain.

ETH, BTC; BCH and LTC are already behaving like that. 

I added the same behaviour to RUNE and BNB. adjusted the clients cosmos, thorchain and binance

### CODE REVIEW

File: packages/xchain-thorchain/src/client.ts [hunk 1]
```diff
@@ -282,15 +282,24 @@ class Client implements ThorchainClient, XChainClient {
...
+      let assetBalances = balances.map((balance) => ({
+        asset: (balance.denom && getAsset(balance.denom)) || AssetRune,
+        amount: baseAmount(balance.amount, DECIMAL),
+      }))
+
+      if (assetBalances.length === 0) {
+        assetBalances = [
+          {
+            asset: AssetRune,
+            amount: baseAmount('0.0', DECIMAL),
```
Comment (veado) [inline packages/xchain-thorchain/src/client.ts:295]: [nitpick] `baseAmount(0, DECIMAL)`

File: packages/xchain-cosmos/src/client.ts [hunk 2]
```diff
@@ -243,17 +243,26 @@ class Client implements CosmosClient, XChainClient {
...
+          amount: baseAmount(balance.amount, DECIMAL),
+        }
+      })
+
+      // make sure we always have the main asset as balance in the array
+      if (assetBalances.length === 0) {
+        assetBalances = [
+          {
+            asset: mainAsset,
+            amount: baseAmount('0.0', DECIMAL),
```
Comment (veado) [inline packages/xchain-cosmos/src/client.ts:258]: [nitpick] `baseAmount(0, DECIMAL)`

File: packages/xchain-cosmos/__tests__/client.test.ts [hunk 3]
```diff
@@ -129,8 +129,10 @@ describe('Client Test', () => {
       result: [],
     })
 
-    const result = await cosmosClient.getBalance(address0_mainnet)
-    expect(result).toEqual([])
+    const balances = await cosmosClient.getBalance(address0_mainnet)
+    const expected = balances[0].amount.amount().isEqualTo(baseAmount(0, 6).amount())
+    expect(expected).toBeTruthy()
+    expect(balances[0].asset).toEqual(AssetAtom)
```
Comment (veado) [inline packages/xchain-cosmos/__tests__/client.test.ts:135]: To be safe: 
```ts
expect(assetToString(balances[0].asset)).toEqual(assetToString(AssetAtom))
```

File: packages/xchain-thorchain/__tests__/client.test.ts [hunk 4]
```diff
@@ -167,8 +167,10 @@ describe('Client Test', () => {
       height: 0,
       result: [],
     })
-    const result = await thorClient.getBalance(thorClient.getAddress(0))
-    expect(result).toEqual([])
+    const balances = await thorClient.getBalance(thorClient.getAddress(0))
+    expect(balances.length).toEqual(1)
+    expect(balances[0].asset).toEqual(AssetRune)
```
Comment (veado) [inline packages/xchain-thorchain/__tests__/client.test.ts:172]: To be safe: 
```ts
expect(assetToString(balances[0].asset)).toEqual(assetToString(AssetRune))
```

File: packages/xchain-binance/src/client.ts [hunk 5]
```diff
@@ -244,17 +244,26 @@ class Client implements BinanceClient, XChainClient {
...
+          amount: assetToBase(assetAmount(balance.free, 8)),
+        }
+      })
+
+      // make sure we always have the bnb asset as balance in the array
+      if (assetBalances.length === 0) {
+        assetBalances = [
+          {
+            asset: AssetBNB,
+            amount: assetToBase(assetAmount('0.0', 8)),
```
Comment (veado) [inline packages/xchain-binance/src/client.ts:259]: Could you add `const BNB_DECIMAL = 8` to 'xchainjs-lib/packages/xchain-binance/src/utils.ts` (similar to other packages) to use it for example as follow here ^:

`assetToBase(assetAmount('0.0', 8))` -> `baseAmount(0, BNB_DECIMAL)`

File: packages/xchain-binance/__tests__/client.test.ts [hunk 6]
```diff
@@ -166,7 +171,13 @@ describe('BinanceClient Test', () => {
...
     )
     balances = await bnbClient.getBalance('bnb1ja07feunxx6z9kue3fn05dazt0gpn4y9e5t8rn')
-    expect(balances).toEqual([])
+
+    expect(balances.length).toEqual(1)
+
+    amount = balances[0].amount
+
+    expect(amount.amount().isEqualTo(0)).toBeTruthy()
+    expect(balances[0].asset.ticker === 'BNB').toBeTruthy()
```
Comment (veado) [inline packages/xchain-binance/__tests__/client.test.ts:180]: Same here?
```ts
expect(assetToString(balances[0].asset)).toEqual(assetToString(AssetBNB))
```

File: packages/xchain-binance/__tests__/client.test.ts [hunk 7]
```diff
@@ -150,7 +150,12 @@ describe('BinanceClient Test', () => {
       sequence: 0,
     })
     let balances = await bnbClient.getBalance('bnb1v8cprldc948y7mge4yjept48xfqpa46mmcrpku')
-    expect(balances).toEqual([])
+    expect(balances.length).toEqual(1)
+
+    let amount = balances[0].amount
+
+    expect(amount.amount().isEqualTo(0)).toBeTruthy()
+    expect(balances[0].asset.ticker === 'BNB').toBeTruthy()
```
Comment (veado) [inline packages/xchain-binance/__tests__/client.test.ts:158]: Maybe this?
```ts
expect(assetToString(balances[0].asset)).toEqual(assetToString(AssetBNB))
```

**Review by veado — CHANGES_REQUESTED** [review]
@severinwullschleger LGTM, just few (small) suggestions ...

**Review by veado — DISMISSED** [review]

**Comment by veado** [comment]
@severinwullschleger Last change request: Pls bump versions in `package.json` of each `xchain-*` packages you have updated and add a note in CHANGELOG? After bumping + merging your updates can be published to npm then. Thx!

**Comment by severinwullschleger** [comment]
@veado did that

**Comment by severinwullschleger** [comment]
@veado fixed the conflicts, can you merge it?

**Comment by matikucharski** [comment]
Is it going to be merged?

**Comment by veado** [comment]
> Is it going to be merged?

Conflicts need to be fixed before going to merge, anyone?

**Pull request closed without merging**